### DISCUSSION TITLE
Hardware rendering on Wayland with new 545 NVidia driver

### DISCUSSION THREAD

**Discussion by ArtemSmaznov** [qutebrowser/qutebrowser#7996] [General]
**Version info**:
qutebrowser v3.0.2
Git commit: 
Backend: QtWebEngine 6.6, based on Chromium 112.0.5615.213 (from api)
Qt: 6.6.0

CPython: 3.11.5
PyQt: 6.6.0

Qt wrapper info:
  PyQt6: success
  PyQt5: not imported
  -> selected: PyQt6 (via autoselect)

colorama: 0.4.6
jinja2: 3.1.2
pygments: 2.16.1
yaml: 6.0.1
adblock: 0.6.0
objc: no
PyQt6.QtWebEngineCore: 6.6.0
PyQt6.sip: 6.7.12
pdf.js: 3.11.174 (/usr/share/pdf.js/build/pdf.js)
sqlite: 3.44.0
QtNetwork SSL: OpenSSL 3.1.4 24 Oct 2023

Style: Qt6CTProxyStyle
Platform plugin: wayland
OpenGL: OpenGL ES
Platform: Linux-6.6.1-zen1-1-zen-x86_64-with-glibc2.38, 64bit
Linux distribution: Arch Linux (arch)
Frozen: False
Imported from /usr/lib/python3.11/site-packages/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/qt6, data path: /usr/share/qt6

**Does the bug happen if you start with `--temp-basedir`?**:
yes 

**Description**
This is likely going to be invalid as a bug in the driver but I am trying to find out what is the issue.
I have been using qutebrowser just fine on Hyprland having set `c.qt.force_software_rendering = 'none'`
After I have updated my NVidia driver 535->545 I am experiencing serious lag when viewing pages: I would load the page, then be able to interact with it (scrolling, selecting, etc) for a second or two and then it would freeze for ~30sec before allowing me another second or two to interact again. Affects all pages as far as I can tell.
I have set it to other options essentially disabling hardware rendering and it is better on most sites but some are quite laggy (same as on 535 driver I think).
Brave, Firefox and Nyxt seem to be be quite fluid though. Qutebrowser seems to be working fine on X though but I have not really used X outside of testing this.
I am curious if any of the people here might have a better idea how low level driver/graphical stuff.
I have talked to people in the Hyprland community and they seem confident it is a driver issue but seggested to check with qutebrowser as well.

**How to reproduce**
gpu driver: nvidia-dkms 535
wayland compositor: hyprland-nvidia-git
c.qt.force_software_rendering = 'none'
1. open any page (github for example)
2. scroll the page a bunch, select random text, etc
3. all is snappy
4. update nvidia-dkms to 545 and reboot
5. retry step 2
6. page is practically unusable - even typing this issue comes with a sever delay of text appearing on screen.

**Comment by The-Compiler** [comment]
Indeed there doesn't seem anything actionable for qutebrowser here - converting this to a discussion instead.

**Reply by ArtemSmaznov** [reply]
Apologies, was not familiar with discussions.
Do you have any idea why qutebrowser is potentially the only program on my system to suffer from a driver update like this?
I am hoping it's just a bug in the driver and will be fixed by NVidia but I understand low level stuff so a bit worried that it might not go away.

**Reply by The-Compiler** [reply]
No clue I'm afraid. QtWebEngine does a bit fancy stuff when it comes to integrating the Chromium rendering stack with Qt's, so it might just be that it is a bit exotic. But you'd almost certainly see the same issue with [the testbrowser](https://github.com/qutebrowser/qutebrowser/blob/main/scripts/testbrowser/testbrowser_webengine.py) as well (make sure it uses PyQt6).

**Reply by ArtemSmaznov** [reply]
Makes sense I guess, I didn't think that you guys abstract from that by using the web engines.
I'll try updating the drivers again and checking the testbrowser out on the weekend

**Reply by ArtemSmaznov** [reply]
Didn't get around trying the test browser you mentioned. I have recently updated my GPU to an AMD one and this is no longer an issue for me so you can close the tread.
